core/lib/zksync_core/src/api_server/web3/mod.rs

```diff
@@ -265,16 +280,11 @@ impl ApiBuilder {
     }
 }
 
-impl ApiBuilder {
-    pub async fn build(
-        self,
-        stop_receiver: watch::Receiver<bool>,
-    ) -> anyhow::Result<ApiServerHandles> {
-        self.into_full_params()?.spawn_server(stop_receiver).await
+impl ApiServer {
+    pub fn health_check(&self) -> ReactiveHealthCheck {
+        self.health_check.clone()
     }
```

Comment:
Storing `health_check` is redundant; it can be instantiated from `health_updater` using `subscribe()` method.